tests/groups/test_mongo.py

```diff
@@ -17,6 +17,11 @@ async def test_update_member_users_and_api_keys(
     user = await fake2.users.create(groups=[group_1, group_2])
 
     async with AsyncSession(pg) as session:
+        await session.execute(
+            delete(user_group_associations).where(
+                user_group_associations.c.group_id == group_2.id
+            )
+        )
         await session.execute(delete(SQLGroup).where(SQLGroup.id == group_2.id))
```

Comment:
I think there is a way to make it so when the delete the group, the rows in the assocation table are automatically removed a well.

It may just work out of the box with SQLAlchemy.

Otherwise, take a look at cascading (`CASCADE`).

Reply:
Fixed by adding ondelete to the foreign key